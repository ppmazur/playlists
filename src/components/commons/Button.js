import styled from 'styled-components';

const Button = styled.div`
  padding: 10px;
  cursor: pointer;
  &:hover {
    background-color: #00979c;
  }
`;

export default Button;
