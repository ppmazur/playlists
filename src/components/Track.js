import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { openPlaylistsModal } from '../actions';
import { parseDuration } from '../helpers/track';
import Button from './commons/Button';

const StyledTrack = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px;
  font-size: medium;
`;

const Artist = styled.span`
  font-weight: bold;
`;

const TrackName = styled.span`
  font-style: italic;
`;

const Duration = styled.span`
  font-weight: lighter;
`;

export const Track = ({ index, openPlaylistsModal, track }) =>
  <StyledTrack>
    {index})
    <Artist>{track.artist}</Artist>
    <TrackName>{track.title}</TrackName>
    <Duration>{parseDuration(track.duration)}</Duration>
    <Button
      onClick={() => openPlaylistsModal(track.id)}
    >
      &#10010;
    </Button>
  </StyledTrack>;

Track.propTypes = {
  index: PropTypes.number,
  openPlaylistsModal: PropTypes.func.isRequired,
  track: PropTypes.shape({
    artist: PropTypes.string.isRequired,
    duration: PropTypes.number.isRequired,
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
};

const mapStateToProps = (state, { trackId }) => ({
  track: state.tracks.byId[trackId],
});

const mapDispatchToProps = dispatch => ({
  openPlaylistsModal: trackId => dispatch(openPlaylistsModal(trackId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Track);
