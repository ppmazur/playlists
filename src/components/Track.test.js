import React from 'react';
import renderer from 'react-test-renderer';

import { Track } from './Track';
import Button from './commons/Button';

describe('<Track />', () => {
  const mockedTrack = {
    artist: 'artist name',
    duration: 123123,
    id: 'id1',
    title: 'title test',
    tracks: [],
  };
  const openModal = jest.fn();
  beforeEach(() => {
    openModal.mockClear();
  });

  it('matches the Track snapshot', () => {
    const result = renderer
      .create(<Track
        openPlaylistsModal={openModal}
        track={mockedTrack}
      />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('calls openPlaylistModal on button click', () => {
    const result = renderer
      .create(<Track
        openPlaylistsModal={openModal}
        track={mockedTrack}
      />);
    result.root.findByType(Button).props.onClick();
    expect(openModal).toHaveBeenCalledWith(mockedTrack.id);
  });
});
