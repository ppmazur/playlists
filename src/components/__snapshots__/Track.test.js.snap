// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> matches the Track snapshot 1`] = `
<div
  className="sc-bwzfXH caIDzr"
>
  )
  <span
    className="sc-htpNat ixiyzb"
  >
    artist name
  </span>
  <span
    className="sc-bxivhb fSOxbl"
  >
    title test
  </span>
  <span
    className="sc-ifAKCX fqUeIU"
  >
    2052:03
  </span>
  <div
    className="sc-bdVaJa cJstJA"
    onClick={[Function]}
  >
    âœš
  </div>
</div>
`;
